geode-unsafe/build.gradle

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply from: "${rootDir}/${scriptDir}/standard-subproject-configuration.gradle"
+apply from: "${project.projectDir}/../gradle/publish.gradle"
+
+dependencies {
+  compile(platform(project(':boms:geode-all-bom')))
+}
+
+ext.findJavac = { ->
```

Comment:
What is this findJavac function for? Isn't gradle already doing this?

Reply:
Gradle doesn't know where the Java command line compiler is. It uses the Java compiler APIs to invoke the compiler. You you want use the command line compiler then you have to provide the path to the executable. In our CI this executable in not in in the path because we have multiple versions of the JDK installed. We need to use forked compile here in order to send the flags to ignore internal APIs. You can't pass those flags to the compiler via the compiler API.

Reply:
Setting the executable is not standard. Shouldn't this use https://docs.gradle.org/current/userguide/building_java_projects.html#example_configure_java_7_build instead?

Reply:
Seems to effectively be doing the same thing. The link expects to use a different JDK than the one running Gradle and have it provided in a property. My approach here uses the Gradle JDK location to find the java executable. The link sets the executable to one of the java executables depending on task type.

Reply:
Ok, I see some magic in there. Let me see if I can do some of the same.